test/unit/ajax.js

```diff
@@ -837,6 +837,92 @@ QUnit.module( "ajax", {
 		};
 	} );
 
+	ajaxTest( "jQuery.ajax() - do not execute scripts from unsuccessful responses (gh-4250)", 7, function( assert ) {
+		var globalEval = jQuery.globalEval;
+
+		function request( title, options, overrides ) {
+			var testMsg = title + ": expected file missing status";
+			return jQuery.extend( options, {
+				beforeSend: function() {
+					jQuery.globalEval = function() {
+						assert.ok( false, "Should not eval" );
+					};
+				},
+				complete: function() {
+					jQuery.globalEval = globalEval;
+				},
+				// error is the significant assertion
+				error: function( xhr ) {
+					assert.strictEqual( xhr.status, 404, testMsg );
+				},
+				success: function() {
+					assert.ok( false, "Unanticipated success" );
+				}
+			}, overrides );
+		}
+
+		return [
+			request(
+				"HTML reply",
+				{
+					url: url( "404.js" )
+				}
+			),
+			request(
+				"HTML reply with dataType",
+				{
+					dataType: "script",
+					url: url( "404.js" )
+				}
+			),
+			request(
+				"script reply",
+				{
+					url: url( "mock.php?action=errorWithScript" )
+				}
+			),
+			request(
+				"script reply with dataType",
+				{
+					dataType: "script",
+					url: url( "mock.php?action=errorWithScript" )
```

Comment:
The test with an explicit `dataType: "script"` should request a URL that doesn't return a JavaScript content type to make sure it's still ignored then.

Reply:
@mgol Just for clarification... I should edit these tests to return a non-JS content type ?  Or I should add tests so that both JS and non-JS content types are returned by the server?

Reply:
@skrobinson As far as I remember, I was thinking about replacing this test but I guess having both variants would be good.